AssertionError: The `python-multipart` library must be installed to use form parsing.

**Expected behavior**
Accessing form data using `await req.form()` should return the form data without errors.

**Actual behavior**
An `AssertionError` is raised by Starlette because `python-multipart` is not installed.

**To Reproduce**

```python
from bocadillo import API, view

api = API()

@api.route("/")
@view(methods=["post"])
async def index(req, res):
    await req.form()

api.client.post("/", data={"foo": "bar"})
```

**Screenshots/Traceback**

```
../../../.local/share/virtualenvs/bocadillo-AaIIUH0P/lib/python3.7/site-packages/requests/sessions.py:581: in post
    return self.request('POST', url, data=data, json=json, **kwargs)
../../../.local/share/virtualenvs/bocadillo-AaIIUH0P/lib/python3.7/site-packages/starlette/testclient.py:382: in request
    json=json,
../../../.local/share/virtualenvs/bocadillo-AaIIUH0P/lib/python3.7/site-packages/requests/sessions.py:533: in request
    resp = self.send(prep, **send_kwargs)
../../../.local/share/virtualenvs/bocadillo-AaIIUH0P/lib/python3.7/site-packages/requests/sessions.py:646: in send
    r = adapter.send(request, **kwargs)
../../../.local/share/virtualenvs/bocadillo-AaIIUH0P/lib/python3.7/site-packages/starlette/testclient.py:211: in send
    raise exc from None
../../../.local/share/virtualenvs/bocadillo-AaIIUH0P/lib/python3.7/site-packages/starlette/testclient.py:208: in send
    loop.run_until_complete(connection(receive, send))
../../../.pyenv/versions/3.7.2/lib/python3.7/asyncio/base_events.py:584: in run_until_complete
    return future.result()
bocadillo/applications.py:320: in dispatch_http
    self.server_error_middleware.raise_if_exception()
bocadillo/errors.py:107: in raise_if_exception
    raise self.exception from None
bocadillo/errors.py:111: in __call__
    res = await self.app(req, res)
bocadillo/errors.py:154: in __call__
    raise exc from None
bocadillo/errors.py:150: in __call__
    res = await self.app(req, res)
bocadillo/routing.py:314: in __call__
    await match.route(req, res, **match.params)
bocadillo/routing.py:224: in __call__
    await handler(req, res, **params)  # type: ignore
tests/test_request.py:108: in index
    form = await req.form()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <bocadillo.request.Request object at 0x110edc240>

    async def form(self) -> FormData:
        if not hasattr(self, "_form"):
            assert (
                parse_options_header is not None
>           ), "The `python-multipart` library must be installed to use form parsing."
E           AssertionError: The `python-multipart` library must be installed to use form parsing.

../../../.local/share/virtualenvs/bocadillo-AaIIUH0P/lib/python3.7/site-packages/starlette/requests.py:182: AssertionError
```

**Possible solutions**
This is quite surprising and not referenced in [the docs](https://bocadilloproject.github.io/guides/http/requests.html#body). We probably don't want users to have to install the dependency themselves considering how small yet possibly common the feature is.

So we should probably just add [python-multipart](https://github.com/andrew-d/python-multipart) as a dependency: add `"python-multipart"` to `install_requires` in `setup.py`.

**Specs**
- OS + Version: macOS High Sierra
- Python Version: 3.7.2
- Bocadillo Version: 0.11.1

**Additional context**
/

--- a/docs/guides/http/requests.md
+++ b/docs/guides/http/requests.md
@@ -55,8 +55,7 @@ req.headers['X-App']  # 'Bocadillo'
 
 ## Query parameters
 
-Query parameters are available at `req.query_params`, an immutable Python
-dictionary-like object.
+Query parameters are available at `req.query_params`, an immutable `MultiDict`, i.e. a dictionary-like object that supports having multiple elements per key.
 
 ```bash
 curl "http://localhost:8000?add=1&sub=2&sub=3"
@@ -87,9 +86,13 @@ only be used inside **asynchronous** views.
 
 You can retrieve it in several ways, depending on the expected encoding:
 
-- Bytes: `await req.body()`
-- Form data: `await req.form()`
-- JSON: `await req.json()`
+| Body type | Invocation         | Return type      |
+| --------- | ------------------ | ---------------- |
+| Raw       | `await req.body()` | `bytes`          |
+| Form data | `await req.form()` | `MultiDict`\*    |
+| JSON      | `await req.json()` | `list` or `dict` |
+
+\*This is a dictionary-like object that behaves like [`query_params`](#query-parameters). It contains both form data and multipart (upload) data.
 
 ::: tip How is malformed JSON handled?
 If the request body is not proper JSON, a `400 Bad Request` error response is returned.

--- a/setup.py
+++ b/setup.py
@@ -28,6 +28,7 @@ setuptools.setup(
         "whitenoise",
         "requests",
         "parse",
+        "python-multipart",
         "websockets>=6.0",
     ],
     url=DOCS,

--- a/tests/test_form_parsing.py
+++ b/tests/test_form_parsing.py
@@ -1,0 +1,16 @@
+from starlette.datastructures import FormData
+
+from bocadillo import API
+
+
+def test_form_parsing(api: API):
+    @api.route("/")
+    async def index(req, res):
+        form = await req.form()
+        assert isinstance(form, FormData)
+        # not actually a dict, so not immediately JSON-serializable
+        res.media = dict(form)
+
+    r = api.client.get("/", data={"foo": "bar"})
+    assert r.status_code == 200
+    assert r.json() == {"foo": "bar"}

